src/rules/no_extra_boolean_cast.rs

```diff
@@ -373,118 +428,166 @@ mod tests {
-        {
-          col: 4,
-          message: NoExtraBooleanCastMessage::BooleanCall,
-          hint: NoExtraBooleanCastHint::BooleanCall,
-        }
-      ],
-      "while (!Boolean()) { foo() }": [
-        {
-          col: 7,
-          message: NoExtraBooleanCastMessage::BooleanCall,
-          hint: NoExtraBooleanCastHint::BooleanCall,
-        }
-      ],
-      "var foo = Boolean() ? bar() : baz()": [
-        {
-          col: 10,
-          message: NoExtraBooleanCastMessage::BooleanCall,
-          hint: NoExtraBooleanCastHint::BooleanCall,
-        }
-      ],
-      "if (Boolean()) { foo() }": [
-        {
-          col: 4,
-          message: NoExtraBooleanCastMessage::BooleanCall,
-          hint: NoExtraBooleanCastHint::BooleanCall,
-        }
-      ],
-      "while (Boolean()) { foo() }": [
-        {
-          col: 7,
-          message: NoExtraBooleanCastMessage::BooleanCall,
-          hint: NoExtraBooleanCastHint::BooleanCall,
-        }
-      ],
-      "Boolean(Boolean(foo))": [
-        {
-          col: 8,
-          message: NoExtraBooleanCastMessage::BooleanCall,
-          hint: NoExtraBooleanCastHint::BooleanCall,
-        }
-      ],
-      "Boolean(!!foo, bar)": [
-        {
-          col: 8,
-          message: NoExtraBooleanCastMessage::DoubleNegation,
-          hint: NoExtraBooleanCastHint::DoubleNegation,
-        }
-      ],
-      "function *foo() { yield!!a ? b : c }": [
-        {
-          col: 23,
-          message: NoExtraBooleanCastMessage::DoubleNegation,
-          hint: NoExtraBooleanCastHint::DoubleNegation,
-        }
-      ],
-      "function *foo() { yield!! a ? b : c }": [
-        {
-          col: 23,
-          message: NoExtraBooleanCastMessage::DoubleNegation,
-          hint: NoExtraBooleanCastHint::DoubleNegation,
-        }
-      ],
-      "function *foo() { yield! !a ? b : c }": [
-        {
-          col: 23,
-          message: NoExtraBooleanCastMessage::DoubleNegation,
-          hint: NoExtraBooleanCastHint::DoubleNegation,
-        }
-      ],
-      "function *foo() { yield !!a ? b : c }": [
-        {
-          col: 24,
-          message: NoExtraBooleanCastMessage::DoubleNegation,
-          hint: NoExtraBooleanCastHint::DoubleNegation,
-        }
-      ],
-      "function *foo() { yield(!!a) ? b : c }": [
-        {
-          col: 24,
-          message: NoExtraBooleanCastMessage::DoubleNegation,
-          hint: NoExtraBooleanCastHint::DoubleNegation,
-        }
-      ],
-      "function *foo() { yield/**/!!a ? b : c }": [
-        {
-          col: 27,
-          message: NoExtraBooleanCastMessage::DoubleNegation,
-          hint: NoExtraBooleanCastHint::DoubleNegation,
-        }
-      ],
+      "!Boolean((foo, bar()));": {
+        "!(foo, bar());",
+        [
+          {
+            col: 0,
+            message: NoExtraBooleanCastMessage::BooleanCall,
+            hint: NoExtraBooleanCastHint::BooleanCall,
+          }
+        ]
+      },
```

Comment:
This macro may be a bit harder to read, probably because there's no indicator on what `"!(foo, bar());"` means. What about the following syntax, for instance?

```rs
      "!Boolean((foo, bar()));": {
        autofixed: "!(foo, bar());",
        errors: [
          {
            col: 0,
            message: NoExtraBooleanCastMessage::BooleanCall,
            hint: NoExtraBooleanCastHint::BooleanCall,
          }
        ]
      },
```

Reply:
I agree with @magurotuna here - may I suggest to create a completely new macro so we don't overload the existing one further.

I wonder if we should have some kind of a sanity check in the macro to make sure that the proposed output actually passes linting